### PR TITLE
Qol features for drones

### LINKED ISSUE #10928: Drones and gravitygen-less tiles
Unless a drone happens to have a fire extinguisher with them, they're completely fucked if they get stranded on tiles on a level without a gravity gen.

They can't throw anything, anything they construct doesn't update their ability to move, they can't even trip balls over lockers. (I constructed a girder and a grille and a wall on the spot I was floating on and still couldn't move.)

How to test: Go to box wrecked telecomms as a drone, get stranded on a tile or in the middle of the abandoned teleporter without a fire extinguisher. Flail around helplessly no matter what you do for the rest of the round.

### PR DESCRIPTION
**Pull request by Shadowlight213**
Drones are now immune to spacewind and no gravity
Drones can also now set airlock access on RCDs and airlock electronics.
Fixes #10928

### CODE REVIEW

File: code/game/objects/items/weapons/RCD.dm [hunk 1]
```diff
@@ -72,7 +72,7 @@ RCD
 	set category = "Object"
 	set src in usr
 
-	if (!ishuman(usr) && !isrobot(usr))
+	if (!ishuman(usr) && !isrobot(usr) && !isdrone(usr))
```
Comment (GunHog) [inline code/game/objects/items/weapons/RCD.dm]: Why not use !usr.has_unlimited_silicon_privilege? That would let you eliminate the borg check as well.

File: code/game/objects/items/weapons/RCD.dm [hunk 2]
```diff
@@ -133,7 +133,7 @@ RCD
 		return
 
 	if (href_list["login"])
-		if(allowed(usr))
+		if(allowed(usr) || isdrone(usr))
```
Comment (GunHog) [inline code/game/objects/items/weapons/RCD.dm]: I believe ````allowed()```` already automatically accepts drones.

**Comment by bgobandit** [comment]
Fixes https://github.com/tgstation/-tg-station/issues/11598.

**Comment by newfren** [comment]
Holy shit this is amazing the amount of times I've teleported to the abandoned teleporter then realised I didn't have a fire extinguisher or something and uselessly rotated in the middle of the room is disgusting.

**Comment by Shadowlight213** [comment]
Ok so after extensive testing and attempts, I was unable to get the alert window to actually work, so I'm going to have to pull that out of the PR.

**Pull request merged**